File: packages/witness/src/templates/selenium-junit/VisualExampleTest.java
// TestivAI Visual Regression Example
// Docs: https://testiv.ai/docs#java
package testivai;

import org.junit.jupiter.api.*;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

class VisualExampleTest {
    static WebDriver driver;

    @BeforeAll
    static void setup() {
        ChromeOptions options = new ChromeOptions();
        options.addArguments("--remote-debugging-port=9222");
        driver = new ChromeDriver(options);
    }

    @AfterAll
    static void teardown() {
        driver.quit();
    }

    @Test
    void homepageLooksCorrect() {
        driver.get("http://localhost:3000");
        TestivAIWitness.witness(driver, "homepage");
    }
}

File: packages/witness/src/templates/selenium-js/visual-example.test.js
// TestivAI Visual Regression Example
// Docs: https://testiv.ai/docs#selenium
const { Builder } = require('selenium-webdriver');
const chrome = require('selenium-webdriver/chrome');
const { witness } = require('../testivai-witness');

describe('Visual Regression', () => {
  let driver;

  beforeAll(async () => {
    const options = new chrome.Options();
    options.addArguments('--remote-debugging-port=9222');
    driver = await new Builder().forBrowser('chrome').setChromeOptions(options).build();
  });

  afterAll(async () => await driver.quit());

  it('captures homepage', async () => {
    await driver.get('http://localhost:3000');
    await witness(driver, 'homepage');
  });
});
